<?xml version="1.0" encoding="UTF-8"?>
<ticket>
  <assigned-user-id type="integer" nil="true"></assigned-user-id>
  <attachments-count type="integer">1</attachments-count>
  <closed type="boolean">false</closed>
  <created-at type="datetime">2011-03-04T14:33:23+00:00</created-at>
  <creator-id type="integer">68536</creator-id>
  <milestone-due-on type="datetime" nil="true"></milestone-due-on>
  <milestone-id type="integer" nil="true"></milestone-id>
  <number type="integer">6521</number>
  <permalink>problem-running-app-generator-standalone</permalink>
  <priority type="integer">0</priority>
  <project-id type="integer">8994</project-id>
  <raw-data type="binary" nil="true" encoding="base64"></raw-data>
  <state>new</state>
  <tag nil="true"></tag>
  <title>Problem running app generator standalone</title>
  <updated-at type="datetime">2011-03-06T14:09:37+00:00</updated-at>
  <user-id type="integer">68536</user-id>
  <version type="integer">2</version>
  <user-name>Steve Hodgkiss</user-name>
  <creator-name>Steve Hodgkiss</creator-name>
  <url>http://rails.lighthouseapp.com/projects/8994/tickets/6521</url>
  <priority-name nil="true"></priority-name>
  <original-body>I'm trying to use the AppGenerator inside an engine, and the copy_file method is going directly to fileutils instead of using the thor actions method. The error is on master, it works on 3.0.5.

rake generate looks something like this:

    require 'rails/generators'
    require 'rails/generators/rails/app/app_generator'

    class TestAppGenerator &lt; Rails::Generators::AppBase
      source_root File.expand_path(&quot;../templates&quot;, __FILE__)
  
      def run_rails_new
        invoke Rails::Generators::AppGenerator, [&quot;spec/dummy&quot;]
      end
      ...
    end
    TestAppGenerator.start

    % rake generate --trace
    (in /Users/steve/code/engine_test)
    ** Invoke generate (first_time)
    ** Execute generate
           exist  
    rake aborted!
    wrong number of arguments (1 for 2)
    /Users/steve/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/fileutils.rb:471:in `copy_file'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:33:in `readme'
    /Users/steve/code/rails/railties/lib/rails/generators/app_base.rb:85:in `build'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:181:in `create_root_files'
    /Users/steve/.rvm/gems/ruby-1.9.2-p180/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'


Using git diff I've tracked it down to the removal of a bunch of methods being defined in 3.0.5 which are removed in master.

git diff v3.0.5..master railties/lib/rails/generators/rails/app/app_generator.rb

0904e8256864239f673bf91fce1cfffb9345ee61 - This is the commit that stops it working, although I find it strange that it's from October last year and it's not in the 3.0.5 release. Maybe I'm looking at the wrong commit.

In any case adding the method definitions back solves the problem I'm having. I can't figure out why it fails in this case but works fine through the rails command though...

Thanks</original-body>
  <latest-body>I'm trying to use the AppGenerator inside an engine, and the copy_file method is going directly to fileutils instead of using the thor actions method. The error is on master, it works on 3.0.5.

rake generate looks something like this:

    require 'rails/generators'
    require 'rails/generators/rails/app/app_generator'

    class TestAppGenerator &lt; Rails::Generators::AppBase
      source_root File.expand_path(&quot;../templates&quot;, __FILE__)
  
      def run_rails_new
        invoke Rails::Generators::AppGenerator, [&quot;spec/dummy&quot;]
      end
      ...
    end
    TestAppGenerator.start

    % rake generate --trace
    (in /Users/steve/code/engine_test)
    ** Invoke generate (first_time)
    ** Execute generate
           exist  
    rake aborted!
    wrong number of arguments (1 for 2)
    /Users/steve/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/fileutils.rb:471:in `copy_file'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:33:in `readme'
    /Users/steve/code/rails/railties/lib/rails/generators/app_base.rb:85:in `build'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:181:in `create_root_files'
    /Users/steve/.rvm/gems/ruby-1.9.2-p180/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'


Using git diff I've tracked it down to the removal of a bunch of methods being defined in 3.0.5 which are removed in master.

git diff v3.0.5..master railties/lib/rails/generators/rails/app/app_generator.rb

0904e8256864239f673bf91fce1cfffb9345ee61 - This is the commit that stops it working, although I find it strange that it's from October last year and it's not in the 3.0.5 release. Maybe I'm looking at the wrong commit.

In any case adding the method definitions back solves the problem I'm having. I can't figure out why it fails in this case but works fine through the rails command though...

Thanks</latest-body>
  <original-body-html>&lt;div&gt;&lt;p&gt;I'm trying to use the AppGenerator inside an engine, and the
copy_file method is going directly to fileutils instead of using
the thor actions method. The error is on master, it works on
3.0.5.&lt;/p&gt;
&lt;p&gt;rake generate looks something like this:&lt;/p&gt;
&lt;pre&gt;
&lt;code&gt;require 'rails/generators'
require 'rails/generators/rails/app/app_generator'

class TestAppGenerator &amp;lt; Rails::Generators::AppBase
  source_root File.expand_path(&quot;../templates&quot;, __FILE__)

  def run_rails_new
    invoke Rails::Generators::AppGenerator, [&quot;spec/dummy&quot;]
  end
  ...
end
TestAppGenerator.start

% rake generate --trace
(in /Users/steve/code/engine_test)
** Invoke generate (first_time)
** Execute generate
       exist  
rake aborted!
wrong number of arguments (1 for 2)
/Users/steve/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/fileutils.rb:471:in `copy_file'
/Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:33:in `readme'
/Users/steve/code/rails/railties/lib/rails/generators/app_base.rb:85:in `build'
/Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:181:in `create_root_files'
/Users/steve/.rvm/gems/ruby-1.9.2-p180/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'&lt;/code&gt;
&lt;/pre&gt;
&lt;p&gt;Using git diff I've tracked it down to the removal of a bunch of
methods being defined in 3.0.5 which are removed in master.&lt;/p&gt;
&lt;p&gt;git diff v3.0.5..master
railties/lib/rails/generators/rails/app/app_generator.rb&lt;/p&gt;
&lt;p&gt;0904e8256864239f673bf91fce1cfffb9345ee61 - This is the commit
that stops it working, although I find it strange that it's from
October last year and it's not in the 3.0.5 release. Maybe I'm
looking at the wrong commit.&lt;/p&gt;
&lt;p&gt;In any case adding the method definitions back solves the
problem I'm having. I can't figure out why it fails in this case
but works fine through the rails command though...&lt;/p&gt;
&lt;p&gt;Thanks&lt;/p&gt;&lt;/div&gt;</original-body-html>
  <versions type="array">
    <version type="Ticket::Version">
      <assigned-user-id type="integer" nil="true"></assigned-user-id>
      <attachments-count type="integer">0</attachments-count>
      <body>I'm trying to use the AppGenerator inside an engine, and the copy_file method is going directly to fileutils instead of using the thor actions method. The error is on master, it works on 3.0.5.

rake generate looks something like this:

    require 'rails/generators'
    require 'rails/generators/rails/app/app_generator'

    class TestAppGenerator &lt; Rails::Generators::AppBase
      source_root File.expand_path(&quot;../templates&quot;, __FILE__)
  
      def run_rails_new
        invoke Rails::Generators::AppGenerator, [&quot;spec/dummy&quot;]
      end
      ...
    end
    TestAppGenerator.start

    % rake generate --trace
    (in /Users/steve/code/engine_test)
    ** Invoke generate (first_time)
    ** Execute generate
           exist  
    rake aborted!
    wrong number of arguments (1 for 2)
    /Users/steve/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/fileutils.rb:471:in `copy_file'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:33:in `readme'
    /Users/steve/code/rails/railties/lib/rails/generators/app_base.rb:85:in `build'
    /Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:181:in `create_root_files'
    /Users/steve/.rvm/gems/ruby-1.9.2-p180/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'


Using git diff I've tracked it down to the removal of a bunch of methods being defined in 3.0.5 which are removed in master.

git diff v3.0.5..master railties/lib/rails/generators/rails/app/app_generator.rb

0904e8256864239f673bf91fce1cfffb9345ee61 - This is the commit that stops it working, although I find it strange that it's from October last year and it's not in the 3.0.5 release. Maybe I'm looking at the wrong commit.

In any case adding the method definitions back solves the problem I'm having. I can't figure out why it fails in this case but works fine through the rails command though...

Thanks</body>
      <body-html>&lt;div&gt;&lt;p&gt;I'm trying to use the AppGenerator inside an engine, and the
copy_file method is going directly to fileutils instead of using
the thor actions method. The error is on master, it works on
3.0.5.&lt;/p&gt;
&lt;p&gt;rake generate looks something like this:&lt;/p&gt;
&lt;pre&gt;
&lt;code&gt;require 'rails/generators'
require 'rails/generators/rails/app/app_generator'

class TestAppGenerator &amp;lt; Rails::Generators::AppBase
  source_root File.expand_path(&quot;../templates&quot;, __FILE__)

  def run_rails_new
    invoke Rails::Generators::AppGenerator, [&quot;spec/dummy&quot;]
  end
  ...
end
TestAppGenerator.start

% rake generate --trace
(in /Users/steve/code/engine_test)
** Invoke generate (first_time)
** Execute generate
       exist  
rake aborted!
wrong number of arguments (1 for 2)
/Users/steve/.rvm/rubies/ruby-1.9.2-p180/lib/ruby/1.9.1/fileutils.rb:471:in `copy_file'
/Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:33:in `readme'
/Users/steve/code/rails/railties/lib/rails/generators/app_base.rb:85:in `build'
/Users/steve/code/rails/railties/lib/rails/generators/rails/app/app_generator.rb:181:in `create_root_files'
/Users/steve/.rvm/gems/ruby-1.9.2-p180/gems/thor-0.14.6/lib/thor/task.rb:22:in `run'&lt;/code&gt;
&lt;/pre&gt;
&lt;p&gt;Using git diff I've tracked it down to the removal of a bunch of
methods being defined in 3.0.5 which are removed in master.&lt;/p&gt;
&lt;p&gt;git diff v3.0.5..master
railties/lib/rails/generators/rails/app/app_generator.rb&lt;/p&gt;
&lt;p&gt;0904e8256864239f673bf91fce1cfffb9345ee61 - This is the commit
that stops it working, although I find it strange that it's from
October last year and it's not in the 3.0.5 release. Maybe I'm
looking at the wrong commit.&lt;/p&gt;
&lt;p&gt;In any case adding the method definitions back solves the
problem I'm having. I can't figure out why it fails in this case
but works fine through the rails command though...&lt;/p&gt;
&lt;p&gt;Thanks&lt;/p&gt;&lt;/div&gt;</body-html>
      <closed type="boolean">false</closed>
      <created-at type="datetime">2011-03-04T14:33:23+00:00</created-at>
      <creator-id type="integer">68536</creator-id>
      <diffable-attributes type="yaml">--- {}

</diffable-attributes>
      <milestone-id type="integer" nil="true"></milestone-id>
      <number type="integer">6521</number>
      <permalink>problem-running-app-generator-standalone</permalink>
      <priority type="integer">0</priority>
      <project-id type="integer">8994</project-id>
      <state>new</state>
      <tag nil="true"></tag>
      <title>Problem running app generator standalone</title>
      <updated-at type="datetime">2011-03-04T14:34:23+00:00</updated-at>
      <user-id type="integer">68536</user-id>
      <version type="integer">1</version>
      <user-name>Steve Hodgkiss</user-name>
      <creator-name>Steve Hodgkiss</creator-name>
      <url>http://rails.lighthouseapp.com/projects/8994/tickets/6521</url>
      <priority-name nil="true"></priority-name>
    </version>
    <version type="Ticket::Version">
      <assigned-user-id type="integer" nil="true"></assigned-user-id>
      <attachments-count type="integer">1</attachments-count>
      <body>This is due running it with rake, where FileUtils is included into the top level module causing method_missing not to be used on Rails::ActionMethods.

I've put an example on github that demonstrates this - https://github.com/stevehodgkiss/rails-ticket-8994</body>
      <body-html>&lt;div&gt;&lt;p&gt;This is due running it with rake, where FileUtils is included
into the top level module causing method_missing not to be used on
Rails::ActionMethods.&lt;/p&gt;
&lt;p&gt;I've put an example on github that demonstrates this - &lt;a href=
&quot;https://github.com/stevehodgkiss/rails-ticket-8994&quot;&gt;https://github.com/stevehodgkiss/rails-ticket-8994&lt;/a&gt;&lt;/p&gt;&lt;/div&gt;</body-html>
      <closed type="boolean">false</closed>
      <created-at type="datetime">2011-03-06T14:09:36+00:00</created-at>
      <creator-id type="integer">68536</creator-id>
      <diffable-attributes type="yaml">--- {}

</diffable-attributes>
      <milestone-id type="integer" nil="true"></milestone-id>
      <number type="integer">6521</number>
      <permalink>problem-running-app-generator-standalone</permalink>
      <priority type="integer">0</priority>
      <project-id type="integer">8994</project-id>
      <state>new</state>
      <tag nil="true"></tag>
      <title>Problem running app generator standalone</title>
      <updated-at type="datetime">2011-03-06T14:09:37+00:00</updated-at>
      <user-id type="integer">68536</user-id>
      <version type="integer">2</version>
      <user-name>Steve Hodgkiss</user-name>
      <creator-name>Steve Hodgkiss</creator-name>
      <url>http://rails.lighthouseapp.com/projects/8994/tickets/6521</url>
      <priority-name nil="true"></priority-name>
    </version>
  </versions>
  <attachments type="array">
    <attachment type="Attachment">
      <code>beecf1b392e308e63ea808fccf383c3a11d12a3f</code>
      <content-type>text/plain</content-type>
      <created-at type="datetime">2011-03-04T14:33:23+00:00</created-at>
      <filename>revert_delegate_everything_to_m.diff</filename>
      <height type="integer" nil="true"></height>
      <id type="integer">942491</id>
      <size type="integer">1564</size>
      <uploader-id type="integer">68536</uploader-id>
      <width type="integer" nil="true"></width>
      <url>http://rails.lighthouseapp.com/attachments/942491/revert_delegate_everything_to_m.diff</url>
    </attachment>
  </attachments>
</ticket>
